src/box/sql/build.c

```diff
@@ -714,36 +741,12 @@ sql_create_check_contraint(struct Parse *parser)
 	bool is_alter_add_constr = space == NULL;
 
 	/* Prepare payload for ck constraint definition. */
-	struct region *region = &parser->region;
 	struct Token *name_token = &create_ck_def->base.base.name;
-	const char *name;
-	if (name_token->n > 0) {
-		name = sql_normalized_name_region_new(region, name_token->z,
-						      name_token->n);
-		if (name == NULL) {
-			parser->is_aborted = true;
-			return;
-		}
-	} else {
-		assert(!is_alter_add_constr);
-		uint32_t ck_idx =
-			++parser->create_ck_constraint_parse_def.count;
-		/*
-		 * If it is <ALTER TABLE ADD COLUMN> we should
-		 * count the existing CHECK constraints in the
-		 * space and form a name based on this.
-		 */
-		if (parser->create_table_def.new_space == NULL) {
-			struct space *original_space =
-				space_by_name(space->def->name);
-			assert(original_space != NULL);
-			struct ck_constraint *ck;
-			rlist_foreach_entry(ck, &original_space->ck_constraint,
-					    link)
-				ck_idx++;
-		}
-		name = tt_sprintf("ck_unnamed_%s_%u", space->def->name, ck_idx);
-	}
+	char *name;
+	if (name_token->n != 0)
+		name = sql_name_from_token(parser->db, name_token);
+	else
+		name = sql_ck_unique_name_new(parser, is_field_ck);
 	size_t name_len = strlen(name);
```

Comment:
Yo, apparently `name` can be `NULL` here if `sql_ck_unique_name_new()` fails. Might be better to check for `NULL`. Alternatively you might make all sql allocations `panic()` when they can't alloc somewhere inside `sqlMalloc` or whatever. Like `xmalloc()` does.

Reply:
Thank you for the review! I added an `'if` for now. I will replace `malloc()` in  `sqlMalloc()` by `xmalloc()` and drop unnecessary checks for `NULL`, but not in this PR.